src/scheduler/pbs_sched.c

```diff
@@ -164,27 +164,30 @@ on_segv(int sig)
 	int *thid;
 
 	thid = (int *) pthread_getspecific(th_id_key);
-	if (thid != NULL && *thid != 0)
-		pthread_exit(NULL);
 
-	if (num_threads > 1)
-		kill_threads();
+	if (thid == NULL || *thid == 0) {
```

Comment:
I don't understand.  You only want to terminate if thread 0 gets a segv?  What if a global caused the segv?  What if there is massive memory corruption that crosses threads?  Don't we always want to exit on a segv?

Reply:
So, this code just says that if I'm thread 0, do some cleanup before I die. If you look at the 'else' part of this if, we do a pthread_exit(), that's the code that the other threads will execute if they get the segv. Does that make sense?

Reply:
SEGV should go to all threads in a process, not just the one which caused it right?

Reply:
I saw the pthread_exit().  That doesn't kill the whole process though.  Why aren't we dying on all segvs, not just the ones that happen in thread 1?  Or is it that this handler is called N times for each of the N threads?